lib/Runtime/Library/TypedArray.h

```diff
@@ -415,10 +438,28 @@ namespace Js
             return FALSE;
         }
 
+        inline BOOL BaseTypedDirectSetItemNoDetachCheck(__in uint32 index, __in Js::Var value, TypeName(*convFunc)(Var value, ScriptContext* scriptContext))
+        {
+            TypeName typedValue = convFunc(value, GetScriptContext());
+
+            // The caller of the function made sure that no IsDetached check required.
+            // The caller of the function also made sure that no length check required.
+
+            AssertMsg(index < GetLength(), "Trying to set out of bound index for typed array.");
+            Assert((index + 1)* sizeof(TypeName) + GetByteOffset() <= GetArrayBuffer()->GetByteLength());
```

Comment:
Right, but with assert we will make sure that callers from newer code paths in future has the Detached check.

---
In reply to: [85444234](https://github.com/Microsoft/ChakraCore/pull/1839#discussion_r85444234) [](ancestors = 85444234,85444051)